lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -179,80 +171,113 @@ public void unsubscribe() {
 	 *
 	 */
 
+	/* End sync and emit leave messages for residual members */
+	private void endSyncAndEmitLeaves() {
+		currentSyncChannelSerial = null;
+		List<PresenceMessage> residualMembers = presence.endSync();
+		for (PresenceMessage member: residualMembers) {
+			/*
+			 * RTP19: ... The PresenceMessage published should contain the original attributes of the presence
+			 * member with the action set to LEAVE, PresenceMessage#id set to null, and the timestamp set
+			 * to the current time ...
+			 */
+			member.action = PresenceMessage.Action.leave;
+			member.id = null;
+			member.timestamp = System.currentTimeMillis();
+		}
+		broadcastPresence(residualMembers.toArray(new PresenceMessage[residualMembers.size()]));
+
+		/**
+		 * (RTP5c2) If a SYNC is initiated as part of the attach, then once the SYNC is complete,
+		 * all members not present in the PresenceMap but present in the internal PresenceMap must
+		 * be re-entered automatically by the client using the clientId and data attributes from
+		 * each. The members re-entered automatically must be removed from the internal PresenceMap
+		 * ensuring that members present on the channel are constructed from presence events sent
+		 * from Ably since the channel became ATTACHED
+		 */
+		if (syncAsResultOfAttach) {
+			syncAsResultOfAttach = false;
+			for (PresenceMessage item: internalPresence.values()) {
+				if (presence.put(item)) {
+						/* Message is new to presence map, send it */
+					try {
+						PresenceMessage itemToSend = (PresenceMessage)item.clone();
+						itemToSend.action = PresenceMessage.Action.enter;
+						updatePresence(itemToSend, new CompletionListener() {
+							@Override
+							public void onSuccess() {
+							}
+
+							@Override
+							public void onError(ErrorInfo reason) {
+									/*
+									 * (RTP5c3) If any of the automatic ENTER presence messages published in RTP5c2
+									 * fail, then an UPDATE event should be emitted on the channel, with a reason
+									 * set to the ErrorInfo received from realtime
+									 */
+								Log.e(TAG, String.format("Cannot automatically re-enter channel %s", channel.name));
+								channel.emitUpdate(reason, false);
```

Comment:
This is an outdated line of code, not sure how you managed to comment on it. Now only client_id is missing